### ISSUE TITLE
A question about the performance of your GDN on HAI.

### ISSUE THREAD

**Issue by zdc132** [sihohan/FuSAGNet#5]
Thank you for your excellent works. However, I have a question about the performance of your GDN on HAI. In your paper, the GDN achieved a precision of 88.37, recall of 60.32, and F1 score of 71.70. I also used the GDN network (from Deng et al.), but my precision is only 80, recall is 0.24, and F1 score is only 0.37. Could you please clarify if you made any modifications to the provided source code of the GDN paper? If so, what modifications were made? If not, could you please provide the network parameters used in your experiments? I intend to email the authors of the paper for clarification

**Comment by seansihohan** [comment]
Hi @zdc132,

I haven't made any modifications to the structure of GDN, but I've played around with the hyperparameters as well as weight initialization methods a bit. I don't have the hyperparameters used, but given that GDN's performance already vary depending on the random seed (which affects weight initializations in multiple parts of the model), you might also want to try and modify the random seed first, and check how the performance changes.

**Comment by zdc132** [comment]
> Hi @zdc132,
> 
> I haven't made any modifications to the structure of GDN, but I've played around with the hyperparameters as well as weight initialization methods a bit. I don't have the hyperparameters used, but given that GDN's performance already vary depending on the random seed (which affects weight initializations in multiple parts of the model), you might also want to try and modify the random seed first, and check how the performance changes.

Thank you for your reply!!! I can try change the random seed first,and, i guess  the hai dataset you import to your network is hai 21.03.I want to reproduce the experimental results on the HAI-21.03 dataset,using the improved network architecture from your paper.So can you provide a general set of hyperparameters you used.Such as
batch-size=32 
epoch=50
silde-win=5
embedding-dim=64
slide-stride=1
random-seed=-999
out-layer-inter-dim=64
out-layer-num=1
val-ratio=0.2
topk=15
lr=1e-3

I think epoch=50 silde-win=5 slide-stride=1 out-layer-num=1 val-ratio=0.2 topk=15 lr=1e-3。These parameters can be unchanged. I should adjust these four parameters:embedding-dim batch-size random-seed out-layer-inter-dim

**Comment by zdc132** [comment]
> Hi @zdc132,
> 
> I haven't made any modifications to the structure of GDN, but I've played around with the hyperparameters as well as weight initialization methods a bit. I don't have the hyperparameters used, but given that GDN's performance already vary depending on the random seed (which affects weight initializations in multiple parts of the model), you might also want to try and modify the random seed first, and check how the performance changes.

Thank you for your reply!!! I can try change the random seed first,and, i guess the hai dataset you import to your network is hai 21.03.I want to reproduce the experimental results on the HAI-21.03 dataset,using the improved network architecture FuSAGNet from your paper.So can you provide a general set of hyperparameters you used.Such as
batch-size=32
epoch=50
silde-win=5
embedding-dim=64
slide-stride=1
random-seed=-999
out-layer-inter-dim=64
out-layer-num=1
val-ratio=0.2
topk=15
lr=1e-3

I think epoch=50 silde-win=5 slide-stride=1 out-layer-num=1 val-ratio=0.2 topk=15 lr=1e-3。These parameters can be unchanged. I should adjust these four parameters:embedding-dim batch-size random-seed out-layer-inter-dim

**Comment by zdc132** [comment]
Hello, as the original author.Based on GDN network,I have made changes to the random seed and experimented with different values ranging from 0 to 40. However, the best F1 score I obtained was only 0.4, which is not a significant improvement compared to my previous training results. The parameters I used  are as follows:
seed = 0~40
BATCH_SIZE = 16
SLIDE_WIN = 5
dim = 64
out_layer_num = 1
slide_stride= 1
topk = 15
out_layer_inter_dim = 128
val_ratio = 0.1
EPOCH = 50
report = 'best'
Is the data set preprocessing method you used DENG's method?

**Comment by zdc132** [comment]
Hello, as the original author.Based on GDN network,I have made changes to the random seed and experimented with different values ranging from 0 to 40. However, the best F1 score I obtained was only 0.4, which is not a significant improvement compared to my previous training results. The parameters I used are as follows:
seed = 0~40
BATCH_SIZE = 16
SLIDE_WIN = 5
dim = 64
out_layer_num = 1
slide_stride= 1
topk = 15
out_layer_inter_dim = 128
val_ratio = 0.1
EPOCH = 50
report = 'best'
Where do you think the parameters I use could be improved？Is the data set preprocessing method you used DENG's method?

**Comment by zdc132** [comment]
Thank you for your reply!!!I guess the hai dataset you import to your network is hai 21.03.I want to reproduce the experimental results on the HAI-21.03 dataset,using the improved network architecture FuSAGNet from your paper.So can you provide a general set of hyperparameters you used.These are parameters I used:
batch-size=32
epoch=50
silde-win=5
embedding-dim=64
slide-stride=1
random-seed=-999
out-layer-inter-dim=64
out-layer-num=1
val-ratio=0.2
topk=15
lr=1e-3
But the rusults are very low,even the precision is no more than 1%

**Comment by zdc132** [comment]
Thank you for your reply!!!I guess the hai dataset you import to your network is HAI 21.03.I want to reproduce the experimental results on the HAI-21.03 dataset,using the improved network architecture FuSAGNet from your paper.I used the data in your paper for the dataset hai21.03:
batch-size=64
epoch=30
silde-win=16
embedding-dim=32
slide-stride=1
random-seed=-999
out-layer-inter-dim=32
out-layer-num=1
val-ratio=0.2
topk=15
lr=0.0005
patience=6
The rusults are as follow:
D:\Anaconda3\envs\GDN\python.exe G:/FuSAGNet-main/main.py
[E 1/30] V: 7.4973 | F: 0.0855 | R: 0.3536 | beta*KLD: 0.005163722205907106
[E 2/30] V: 6.9982 | F: 0.0830 | R: 0.3244 | beta*KLD: 0.002387440763413906
[E 3/30] V: 6.4744 | F: 0.0822 | R: 0.3103 | beta*KLD: 0.001323378412052989
[E 4/30] V: 6.3638 | F: 0.0788 | R: 0.3061 | beta*KLD: 0.0009080374147742987
[E 5/30] V: 6.3280 | F: 0.0750 | R: 0.2919 | beta*KLD: 0.0006710526067763567
[E 6/30] V: 5.9401 | F: 0.0795 | R: 0.2930 | beta*KLD: 0.00042849889723584056
[E 7/30] V: 6.6372 | F: 0.0768 | R: 0.2874 | beta*KLD: 0.00035972538171336055
[E 8/30] V: 6.4631 | F: 0.0767 | R: 0.2890 | beta*KLD: 0.0003748621966224164
[E 9/30] V: 7.0391 | F: 0.0746 | R: 0.2853 | beta*KLD: 0.0004097005003131926
[E 10/30] V: 6.8227 | F: 0.0787 | R: 0.2820 | beta*KLD: 0.00029677548445761204
[E 11/30] V: 6.8664 | F: 0.0786 | R: 0.2832 | beta*KLD: 0.00026886313571594656
[E 12/30] V: 6.5886 | F: 0.0739 | R: 0.2815 | beta*KLD: 0.0003325676661916077
F1: 0.2520 | Pr: 0.7150 | Re: 0.1531
As you can see, the results are not very good.So can you provide a general set of hyperparameters you used to reproduce the result in your paper?

**Comment by zdc132** [comment]
Thank you for your reply!!!I guess the hai dataset you import to your network is HAI 21.03.I want to reproduce the experimental results on the HAI-21.03 dataset,using the improved network architecture FuSAGNet from your paper.I used the parameters in your paper for the dataset hai21.03:
batch-size=64
epoch=30
silde-win=16
embedding-dim=32
slide-stride=1
random-seed=-999
out-layer-inter-dim=32
out-layer-num=1
val-ratio=0.2
topk=15
lr=0.0005
patience=6
beta=1(sparse loss weight)
alpha=0.5(forecasting loss weight)
The rusults are as follow:
D:\Anaconda3\envs\GDN\python.exe G:/FuSAGNet-main/main.py
[E 1/30] V: 7.4973 | F: 0.0855 | R: 0.3536 | betaKLD: 0.005163722205907106
[E 2/30] V: 6.9982 | F: 0.0830 | R: 0.3244 | betaKLD: 0.002387440763413906
[E 3/30] V: 6.4744 | F: 0.0822 | R: 0.3103 | betaKLD: 0.001323378412052989
[E 4/30] V: 6.3638 | F: 0.0788 | R: 0.3061 | betaKLD: 0.0009080374147742987
[E 5/30] V: 6.3280 | F: 0.0750 | R: 0.2919 | betaKLD: 0.0006710526067763567
[E 6/30] V: 5.9401 | F: 0.0795 | R: 0.2930 | betaKLD: 0.00042849889723584056
[E 7/30] V: 6.6372 | F: 0.0768 | R: 0.2874 | betaKLD: 0.00035972538171336055
[E 8/30] V: 6.4631 | F: 0.0767 | R: 0.2890 | betaKLD: 0.0003748621966224164
[E 9/30] V: 7.0391 | F: 0.0746 | R: 0.2853 | betaKLD: 0.0004097005003131926
[E 10/30] V: 6.8227 | F: 0.0787 | R: 0.2820 | betaKLD: 0.00029677548445761204
[E 11/30] V: 6.8664 | F: 0.0786 | R: 0.2832 | betaKLD: 0.00026886313571594656
[E 12/30] V: 6.5886 | F: 0.0739 | R: 0.2815 | betaKLD: 0.0003325676661916077
F1: 0.2520 | Pr: 0.7150 | Re: 0.1531
As you can see, the results are not very good.So can you provide a general set of hyperparameters you used to reproduce the result in your paper?
